packages/viz/src/lib/sources/CARTOSource.ts

```diff
@@ -95,10 +95,18 @@ export class CARTOSource extends Source {
     return this._credentials;
   }
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+  /* eslint-disable */
+  // @ts-ignore
+  public async bins(
```

Comment:
Not sure if you are going to keep now this approach after what we talk, but just in case: IMO bins should be calculated outside of any Source, as a result of a calculation method in a Quantiles independent function or class. 

Obviously its calculation depends on an input from Metadata (that will come from the Source or, in the future, from client-side Viewport calculations). 

So I vote for these methods in a Metadata class:
- `source.metadata.getSample(1000)``
- `source.metadata.getField('fieldName')

and a mechanism to inject the metadata instance in the styles builder

Reply:
Done